dockerspawner/dockerspawner.py

```diff
@@ -1173,3 +1172,22 @@ def _fmt(v):
                 v = v["bind"]
             binds[_fmt(k)] = {"bind": _fmt(v), "mode": m}
         return binds
+
+    def _extra_create_kwargs_subs(self):
```

Comment:
Rather than having a method return only extra_create_kwargs, this should probably be implemented as a generic walker of reformatting a potentially deep container:

- walk lists and dicts
- handle docker models objects? Not sure if this is feasible or necessary.
- where strings are seen (always check that it's a string) -> apply `_fmt(s)`